assets/js/controllers/confirmRecoveryPhrase.controller.js

```diff
@@ -2,105 +2,52 @@ angular
   .module('walletApp')
   .controller('ConfirmRecoveryPhraseCtrl', ConfirmRecoveryPhraseCtrl);
 
-function ConfirmRecoveryPhraseCtrl ($scope, $log, Wallet, Alerts, $uibModalInstance, $translate) {
+function ConfirmRecoveryPhraseCtrl ($scope, $uibModalInstance, Wallet, Alerts, modals) {
   $scope.step = 0;
   $scope.offset = 0;
-  $scope.recoveryPhrase = null;
-
-  $scope.lastWordGroup = false;
-
-  $scope.words = [
-    {
-      value: '',
-      actual: '',
-      index: null,
-      error: null
-    }, {
-      value: '',
-      actual: '',
-      index: null,
-      error: null
-    }, {
-      value: '',
-      actual: '',
-      index: null,
-      error: null
-    }, {
-      value: '',
-      actual: '',
-      index: null,
-      error: null
-    }
-  ];
+  $scope.recoveryPhrase = [];
+  $scope.words = [...Array(4)].map(_ => ({}));
+  $scope.addresses = Wallet.legacyAddresses().filter(a => a.active && !a.isWatchOnly && a.balance > 0);
```

Comment:
You could add a method to `payment.js` to determine the threshold. But without a threshold I assume it will never stop nagging the user? Depending on where else we'll show this button/link in the future, that could be annoying.